### PR TITLE
RCT/YJ/get swh equipment type function

### PR DESCRIPTION
**Pull request by yunjoonjung-PNNL**
I'll continue to develop this function after my 2 questions are answered in PR #1405. Thanks!

### CODE REVIEW

File: rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    ----------
+    rmd: dict
+        RMD at RuleSetModelDescription level
+    service_water_heating_equipment_id: str
+        service water heating equipment id
+
+    Returns
+    -------
+    type: A string
+        ex: type = "ELECTRIC_RESISTANCE_INSTANTANEOUS"
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py:34]: I recommend creating a enum class for the return type.

File: rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        service_water_heating_equipment_id,
+        rmd,
+    )
+
+    swh_tank_type = getattr_(
+        service_water_heating_equipment,
+        "service_water_heating_equipment",
+        "tank",
+        "type",
+    )  # this part needs to be corrected.
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py:49]: https://github.com/open229/ruleset-model-description-schema/blob/develop/docs229/ASHRAE229.schema.json#L4480

This is correct, it is tank.
You maybe mixed it with the tanks under Service Water Heating Distribution Object?

File: rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    assert_(
+        fuel_type
+        in [
+            ENERGY_SOURCE.ELECTRICITY,
+            ENERGY_SOURCE.NATURAL_GAS,
+            ENERGY_SOURCE.FUEL_OIL,
+        ],
+        "fuel type must be one of ``ELECTRICITY`, `NATURAL_GAS`, `FUEL_OIL`",
+    )
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py:69]: Great to catch that.

File: rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+    assert_(
+        fuel_type
+        in [
+            ENERGY_SOURCE.ELECTRICITY,
+            ENERGY_SOURCE.NATURAL_GAS,
+            ENERGY_SOURCE.FUEL_OIL,
+        ],
+        "fuel type must be one of ``ELECTRICITY`, `NATURAL_GAS`, `FUEL_OIL`",
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py:68]: Remove the one extra "`"

File: rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+    type = "OTHER"
+    if swh_type == SERVICE_WATER_HEATER.CONVENTIONAL:
+        if swh_tank_type in INSTANTANEOUS_TYPE:
+            type = "INSTANTANEOUS"
+        elif swh_tank_type in STORAGE_TYPE:
+            type = "STORAGE"
+
+    if fuel_type == ENERGY_SOURCE.ELECTRICITY:
+        type = f"ELECTRIC_RESISTANCE_{swh_tank_type}"
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py:79]: Shouldn't this be `type` instead of `swh_tank_type`?

**Review by weilixu — APPROVED** [review]
Approved with minor issues to address

File: rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py (hunk 5 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py:79]: Good catch! Addressed.

File: rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py (hunk 4 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py:68]: Addressed.

File: rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py (hunk 2 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py:49]: Yeah!  I misinterpreted the schema. Thank you Weili.

File: rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py (hunk 1 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py:46]: Addressed.

File: rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        elif swh_tank_type in STORAGE_TYPE:
+            if fuel_type == ENERGY_SOURCE.ELECTRICITY:
+                type = GetSWHEquipmentType.ELECTRIC_RESISTANCE_STORAGE
+            elif fuel_type == ENERGY_SOURCE.NATURAL_GAS:
+                type = GetSWHEquipmentType.GAS_STORAGE
+            elif fuel_type == ENERGY_SOURCE.FUEL_OIL:
+                type = GetSWHEquipmentType.OIL_STORAGE
+    else:
+        if fuel_type == ENERGY_SOURCE.ELECTRICITY:
+            type = GetSWHEquipmentType.ELECTRIC_RESISTANCE_OTHER
```
Comment (Jiarongx-Xie) [inline rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py:101]: [https://github.com/pnnl/ruleset-checking-tool/blob/b35403785a307f96cea5b8a4cd2c73fb28d9cc17/docs/ruleset_functions/get_swh_equipment_type.md#:~:text=set%20the%20type%20to%20OTHER.%20If%20the%20swh_tank_type%20is%20an%20identifiable%20type%2C%20type%20will%20be%20overwritten%20to%20that%20type%3A%20type%20%3D%20%22OTHER%22](url)

**Review by Jiarongx-Xie — COMMENTED** [review]
Thank you, Yun Joon. The implementation looks good to me overall. I have just one minor comment for you to consider.

File: rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py (hunk 6 above)
Comment (weilixu) [reply rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py:101]: That is correct. We do need an `OTHER` type in the types.
So in here it is:
```python
else:
    type = GetSWHEquipmentType.OTHER
```

Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py:101]: Good point! Addressed.

**Pull request merged**